src/Foundation/NSObject2.cs

```diff
@@ -739,6 +768,16 @@ internal void ClearHandle ()
 				} else {
 					NSObject_Disposer.Add (this);
 				}
+			} else {
+				FreeData ();
+			}
+		}
+
+		unsafe void FreeData ()
+		{
+			if (super != IntPtr.Zero) {
+				Marshal.FreeHGlobal (super);
```

Comment:
From the docs:
```
You can use FreeHGlobal to free any memory from the global heap allocated by AllocHGlobal, ReAllocHGlobal, or any equivalent unmanaged API method. If the hglobal parameter is IntPtr.Zero the method does nothing.
```
I think we can remove the need of the if at all, just do the call and set super to be IntPtrZero.